data/transactions/logic/TEAL_opcodes.md

```diff
@@ -865,6 +865,16 @@ When A is a uint64, index 0 is the least significant bit. Setting bit 3 to 1 on
 
 Decodes X using the base64 encoding E. Specify the encoding with an immediate arg either as URL and Filename Safe (`URLEncoding`) or Standard (`StdEncoding`). See <a href="https://rfc-editor.org/rfc/rfc4648.html#section-4">RFC 4648</a> (sections 4 and 5). It is assumed that the encoding ends with the exact number of `=` padding characters as required by the RFC. When padding occurs, any unused pad bits in the encoding must be set to zero or the decoding will fail. The special cases of `\n` and `\r` are allowed but completely ignored. An error will result when attempting to decode a string with a character that is not in the encoding alphabet or not one of `=`, `\r`, or `\n`.
 
+## json_ref r
+
+- Opcode: 0x5d {return type}
+- Pops: *... stack*, {[]byte A}, {[]byte B}
+- Pushes: any
+- return key B's value from json text A
+- LogicSigVersion >= 6
+
+specify the return type with an immediate arg either as JSONInt or JSONString. When B is a nested key of json object A, A.B returns the value of B. Providing a json with keys containing . will return incorrect value. See <a href="https://datatracker.ietf.org/doc/html/rfc7159.html">RFC 7159</a> for additional json specifications
```

Comment:
How confident are we that the Go parsing follows this spec precisely?  There are a lot of little details about trailing commas, unicode escapes, etc that different implementations do different.  If Go's parsing folows it exactly, that's great, because we can say less here, but we need test cases for ever distinction we can think of and if any depart, say so. (Or if the spec is unclear, say so either wy)

Reply:
will it be better to list out the specs in more details based on the test results from jsonspec_test.go?